lib/doorkeeper/oauth/scopes.rb

```diff
@@ -45,20 +45,26 @@ def has_scopes?(scopes)
       end
 
       def +(other)
-        if other.is_a? Scopes
-          self.class.from_array(all + other.all)
-        else
-          super(other)
-        end
+        self.class.from_array(all + other_array(other))
       end
 
       def <=>(other)
         map(&:to_s).sort <=> other.map(&:to_s).sort
       end
 
       def &(other)
-        other_array = other.present? ? other.all : []
-        self.class.from_array(all & other_array)
+        self.class.from_array(all & other_array(other))
+      end
+
+      private
+
+      def other_array(other)
```

Comment:
Can we rename it to something like `to_array`? I think `other_array` is not descriptive name for this method.

Reply:
Done.